packages/ws/src/ws/WebSocketManager.ts

```diff
@@ -199,7 +203,7 @@ export interface ManagerShardEventsMap {
 	];
 }
 
-export class WebSocketManager extends AsyncEventEmitter<ManagerShardEventsMap> {
+export class WebSocketManager extends AsyncEventEmitter<ManagerShardEventsMap> implements AsyncDisposable {
```

Comment:
Should that `implements AsnycDisposable` also be added to mainlib typings of BaseClient then?